File: solutions/0147-insertion-sort-list/solution.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def insertionSortList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        ls=[]
        h=head
        h1=head
        while(h!=None):
            ls.append(h.val)
            h=h.next
        k=0
        ls.sort()
        while(h1!=None):
            h1.val=ls[k]
            h1=h1.next
            k+=1
        return head

File: solutions/0147-insertion-sort-list/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode insertionSortList(ListNode head) {
        List<Integer>ls=new ArrayList<>();
        ListNode h=head;
        ListNode h1=head;
        while(h!=null){
            ls.add(h.val);
            h=h.next;
        }
        int k=0;
        Collections.sort(ls);
        while(h1!=null){
            h1.val=ls.get(k);
            h1=h1.next;
            k+=1;
        }
        return head;
    }
}
